test/new_relic/agent/distributed_tracing/trace_context_cross_agent_test.rb

```diff
@@ -253,7 +253,11 @@ def verify_attributes(test_case_attributes, actual_attributes, event_type)
               %Q(Unexpected #{event_type} attribute "#{key}")
           end
 
-          # TODO: check vendors in test_case_attributes
+          (test_case_attributes['vendors'] || []).each do |k, v|
+            assert_equal v,
+              actual_attributes[k.to_s],
+              %Q(Wrong "#{k}" #{event_type} attribute; expected #{v.inspect}, was #{actual_attributes[k.to_s].inspect})
+          end
```

Comment:
When I run the new test, I see that it there is a `'vendors'` key present only twice.

In those 2 cases, for me `k` is equal to either `'44@nr'`, `'foo'`, or `'bar'`. `v` is always `nil`. The corresponding `actual_attributes[k.to_s]` value is always `nil`. So at least on my machine with a single Ruby version being tested, we're really only ever comparing `nil` with `nil` and getting a false sense of security from it. Am I doing something incorrectly?

My wild guess is that given the `test_case_attributes['vendors']` value of `['foo', 'bar']`, you wanted `k` to end up holding `'foo'` and `v` to end up holding `'bar'`. If that's the case, we'll need to change the iteration syntax.

Reply:
addressed with https://github.com/newrelic/newrelic-ruby-agent/pull/1692